test/sequential/feeLock.API.swapValueThresholds.test.ts

```diff
@@ -0,0 +1,207 @@
+  await sudo.addMGATokens(sudo);
+});
+
+beforeEach(async () => {
+  [testUser1] = setupUsers();
+
+  await setupApi();
+
+  [firstCurrency, secondCurrency] = await Assets.setupUserWithCurrencies(
+    sudo,
+    [defaultCurrencyValue, defaultCurrencyValue],
+    sudo
+  );
+
+  const updateMetadataEvent = await updateFeeLockMetadata(
+    sudo,
+    null,
+    null,
+    thresholdValue,
+    [
+      [MGA_ASSET_ID, true],
+      [firstCurrency, true],
+    ]
+  );
+  await waitSudoOperationSuccess(updateMetadataEvent);
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(firstCurrency, testUser1, defaultCurrencyValue),
+    Assets.mintToken(secondCurrency, testUser1, defaultCurrencyValue),
+    Assets.mintNative(sudo),
+    Sudo.sudoAs(
+      sudo,
+      Xyk.createPool(
+        firstCurrency,
+        defaultPoolVolumeValue,
+        secondCurrency,
+        defaultPoolVolumeValue
+      )
+    )
+  );
+
+  testUser1.addAsset(MGA_ASSET_ID);
+  testUser1.addAsset(firstCurrency);
+  testUser1.addAsset(secondCurrency);
+});
+
+test("gasless- Given a feeLock correctly configured WHEN the user swaps two tokens defined in the thresholds AND swapValue > threshold THEN the extrinsic is correctly submitted AND No locks AND no fees", async () => {
+  await testUser1.addMGATokens(sudo);
+
+  await createPool(
+    testUser1.keyRingPair,
+    firstCurrency,
+    defaultPoolVolumeValue,
+    MGA_ASSET_ID,
+    defaultPoolVolumeValue
+  ).then((result) => {
+    const eventResponse = getEventResultFromMangataTx(result);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+
+  const saleAssetValue = thresholdValue.mul(new BN(2));
+
+  await testUser1.refreshAmounts(AssetWallet.BEFORE);
+  await testUser1.sellAssets(firstCurrency, secondCurrency, saleAssetValue);
+  await testUser1.refreshAmounts(AssetWallet.AFTER);
+
+  const userFirstCurLockedValue = testUser1
+    .getAsset(firstCurrency)
+    ?.amountAfter.reserved!.sub(
+      testUser1.getAsset(firstCurrency)?.amountBefore.reserved!
+    );
+
+  const userSecondCurLockedValue = testUser1
+    .getAsset(secondCurrency)
+    ?.amountAfter.reserved!.sub(
+      testUser1.getAsset(secondCurrency)?.amountBefore.reserved!
+    );
+
+  const userMgaFees = testUser1
+    .getAsset(MGA_ASSET_ID)
+    ?.amountAfter.reserved!.sub(
+      testUser1.getAsset(MGA_ASSET_ID)?.amountBefore.reserved!
+    );
+
+  expect(userFirstCurLockedValue).bnEqual(new BN(0));
+  expect(userSecondCurLockedValue).bnEqual(new BN(0));
+  expect(userMgaFees).bnEqual(new BN(0));
+});
+
+test("gasless- Given a feeLock correctly configured WHEN the user swaps two tokens defined in the thresholds AND the user has enough MGAs AND swapValue < threshold THEN some MGAs will be locked", async () => {
+  const api = getApi();
+
+  await testUser1.addMGATokens(sudo);
+
+  const feeLockAmount = JSON.parse(
+    JSON.stringify(await api.query.feeLock.feeLockMetadata())
+  ).feeLockAmount;
+
+  await createPool(
```

Comment:
I thin kits the second time I see this pool creation, perhaps should It go to the sudo batch operation?

Reply:
done